camel/societies/role_playing.py

```diff
@@ -184,50 +184,51 @@ def __init__(
         else:
             self.critic = None
 
-    def init_chat(self) -> Tuple[AssistantChatMessage, List[ChatMessage]]:
+    def init_chat(self) -> Tuple[BaseMessage, List[BaseMessage]]:
         r"""Initializes the chat by resetting both of the assistant and user
         agents, and sending the system messages again to the agents using
         chat messages. Returns the assistant's introductory message and the
         user's response messages.
 
         Returns:
-            A tuple containing an `AssistantChatMessage` representing the
-            assistant's introductory message, and a list of `ChatMessage`s
+            A tuple containing an `BaseMessage` representing the
+            assistant's introductory message, and a list of `BaseMessage`s
             representing the user's response messages.
         """
         self.assistant_agent.reset()
         self.user_agent.reset()
 
         # Send the system messages again to the agents using chat messages
-        assistant_msg = AssistantChatMessage(
-            role_name=self.assistant_sys_msg.role_name, role="assistant",
+        assistant_msg = BaseMessage.make_assistant_message(
+            role_name=self.assistant_sys_msg.role_name,
             content=(f"{self.user_sys_msg.content}. "
                      "Now start to give me instructions one by one. "
                      "Only reply with Instruction and Input."))
 
-        user_msg = UserChatMessage(role_name=self.user_sys_msg.role_name,
-                                   content=f"{self.assistant_sys_msg.content}")
+        user_msg = BaseMessage.make_user_message(
+            role_name=self.user_sys_msg.role_name,
+            content=f"{self.assistant_sys_msg.content}")
         assistant_response = self.assistant_agent.step(user_msg)
         if assistant_response.terminated or assistant_response.msgs is None:
             raise ValueError(f"Assistant agent terminated unexpectedly. "
                              f"Error info: {assistant_response.info}")
 
         return assistant_msg, assistant_response.msgs
 
-    def process_messages(
+    def make_critic_choice(
```

Comment:
Not sure if we want to change the name since we do not always use critic.

Reply:
the docstring doesn't really provide any insights into the specific of the method. 
Based purely on it, I couldn't understand the purpose of the method, be it either "process_messages" or "make_critic_choice".

Reply:
My general intuition is that methods like `process_messages` should not exist at all. They are if-elif-elif and should be resolved by a polymorphic function or in a method `reduce_to_single` of a class handling choices. I mean, `List[ChatMessage]` must be made a class with a reduction method as its member.